spec/std/socket/udp_socket_spec.cr

```diff
@@ -78,14 +78,32 @@ describe UDPSocket do
                  expect_raises(Socket::Error, "Unsupported IP address family: INET. For use with IPv6 only") do
                    udp.multicast_interface 0
                  end
-                 udp.multicast_interface Socket::IPAddress.new(unspecified_address, 0)
+
+                 begin
+                   udp.multicast_interface Socket::IPAddress.new(unspecified_address, 0)
+                 rescue e : Socket::Error
+                   if e.message =~ /Protocol not available/
```

Comment:
`Socket::Error` is a `SystemError` and that means it has a `os_error : Errno`. Can't you just check a specific `Errno` here? (maybe `ENOPROTOOPT`)

Reply:
So... to be honest I actually never verified whether this reproduces in the self hosted environment, it's carried over from my experiments using GHA hosted runners. Now I did try to reproduce it in the self hosted environment and of course it doesn't. So not technically part of #9438 after all I guess...

The original error is: https://github.com/jhass/crystal/runs/744998680?check_suite_focus=true#step:7:23447

Unfortunately it does not print the errno (name), maybe it should generally...

So yeah, I can invest some time into trying to coax the errno out of the GHA hosted runner environment, or if you say while we have no environment that triggers this (note that quite possibly some distributions package build environments still might in the future), we shouldn't protect against it, I can also at least extract this change from here to unblock this PR.

Please tell me what  you prefer :)

Reply:
If the specs still passes without changes the answer is quite simple :)
Let's revert this then and keep it at hand in case it happens again.

Reply:
FWIW you guessed right: https://github.com/jhass/crystal/runs/831362514?check_suite_focus=true#step:6:74